pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -685,3 +704,38 @@ func getSpotVMOptions(spotVMOptions *v1beta1.SpotVMOptions) (compute.VirtualMach
 	// https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/spot-instances.md#eviction-policies
 	return compute.Spot, compute.Deallocate, billingProfile, nil
 }
+
+func (s *Reconciler) createAvailabilitySet() (string, error) {
+	// Try to find the availability set for the machine location
+	availabilityZones, err := s.availabilityZonesSvc.Get(context.Background(), &availabilityzones.Spec{
+		VMSize: s.scope.MachineConfig.VMSize,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	availabilityZonesSlice, ok := availabilityZones.([]string)
+	if !ok {
+		return "", fmt.Errorf("unexpected type %T", availabilityZones)
+	}
+
+	if len(availabilityZonesSlice) != 0 {
+		klog.V(4).Infof("No availability set needed for %s because availability zones were found", s.scope.Machine.Name)
```

Comment:
What gonna happen in case of single AZ?
Are we sure that there is no zones at all on swiss azure for example?

Reply:
According to RFE description, there should be no zones in swiss azure. I wasn't able to find any documentation before.